src/irc.py

```diff
@@ -139,42 +175,82 @@ def _write(self, path: pathlib.Path, source: str) -> None:
         path.write_text(source, encoding="utf-8")
         logger.info("Pushed file %s", path)
 
-    def _generate_PEM_file_local(self) -> str:
+    def _generate_PEM_file_local(self) -> None:
         """Generate the PEM file content.
 
-        Returns:
-            A string
+        Raises:
+            
         """
-        pass
+        pem_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem ]] || "
+            + "openssl genpkey -out {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem "
+            + "-outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048",
+        ]
+        logger.info("Creating PEM file for IRC bridge.")
+        exec_process = subprocess.run(
+           pem_create_command, shell=True, check=True, capture_output=True)
+        )
+        logger.info("PEM create output: %s.", exec_process.stdout)
 
-    def _generate_conf_local(self) -> str:
-        """Generate the content of the irc configuration file.
+    def _generate_app_registration_local(self) -> str:
+        """Generate the content of the app registration file.
 
         Returns:
             A string
         """
-        pass
+        app_reg_create_command = [
+                "/bin/bash",
+                "-c",
+                f"[[ -f {IRC_BRIDGE_REGISTRATION_PATH} ]] || "
+                f"/bin/node /app/app.js -r -f {IRC_BRIDGE_REGISTRATION_PATH} "
+                f"-u http://localhost:{IRC_BRIDGE_HEALTH_PORT} "
+                f"-c {IRC_BRIDGE_CONFIG_PATH} -l {IRC_BRIDGE_BOT_NAME}",
+            ],
+        )
 
-    def _generate_app_registration_local(self) -> str:
-        """Generate the content of the app registration file.
+    def _eval_conf_local(self) -> str:
+        """Generate the content of the irc configuration file.
 
         Returns:
-            A string
+            A string to write to the configuration file
         """
-        pass
-
-    def handle_new_db_relation_data(self) -> str:
+        db_string = self._handle_new_db_relation_data()
+        matrix_sting = self._handle_new_matrix_relation_data()
+        # we ignore the matrix string for now until we have a plugins interface
+        with open(f"{IRC_BRIDGE_CONFIG_PATH}/config.yaml", "w") as f:
+            data = yaml.load(f)
```

Comment:
`safe_load` ?

Reply:
Aaah, the Loader was SafeLoader and I removed that, sorry.